datafusion/src/scalar.rs

```diff
@@ -171,6 +179,17 @@ impl PartialOrd for ScalarValue {
         // any newly added enum variant will require editing this list
         // or else face a compile error
         match (self, other) {
+            // TODO decimal type, we just compare the values which have the same precision and scale.
```

Comment:
I think the code in this PR is correct  -- two decimal's can be compared if they have the same precision / scale and they can't be compared (returns `None`) if they have different precision/scales).